packages/theme/src/base/types.ts

```diff
@@ -114,3 +114,43 @@ export type BackstageThemeAdditions = {
   page: PageTheme;
   getPageTheme: (selector: PageThemeSelector) => PageTheme;
 };
+
+/**
+ * Custom Typography
+ *
+ * @public
+ */
+export type Typography = {
```

Comment:
We can use `TypographyOptions` from MUI instead right?

Reply:
I've tried that but I just can't seem to resolve this type error:

```text
Argument of type '{ components: { MuiAlert?: { defaultProps?: Partial<AlertProps> | undefined; styleOverrides?: Partial<OverridesStyleRules<keyof AlertClasses, "MuiAlert", Omit<Mui5Theme, "components">>> | undefined; variants?: { ...; }[] | undefined; } | undefined; ... 115 more ...; MuiUseMediaQuery?: { ...; } | undefined; }; palett...' is not assignable to parameter of type 'ThemeOptions'.
  Types of property 'typography' are incompatible.
    Type 'TypographyOptions' is not assignable to type 'TypographyOptions | ((palette: Palette) => TypographyOptions) | undefined'.
      Type 'Partial<Record<Variant, TypographyStyleOptions> & FontStyleOptions>' is not assignable to type 'TypographyOptions'.
        Types of property 'h1' are incompatible.
          Type 'import("/Users/andre.wanlin/repos/forks/backstage/node_modules/@material-ui/core/styles/createTypography").TypographyStyleOptions | undefined' is not assignable to type 'import("/Users/andre.wanlin/repos/forks/backstage/node_modules/@mui/material/styles/createTypography").TypographyStyleOptions | undefined'.
            Type 'TypographyStyleOptions' is not assignable to type 'CSSProperties'.
              Types of property 'appearance' are incompatible.
                Type 'AppearanceProperty | undefined' is not assignable to type 'Appearance | undefined'.
                  Type '"revert-layer"' is not assignable to type 'Appearance | undefined'.
```

Do you want me to push the changes that got me to this point?

Reply:
Yep I see we got a pretty small subset of the available options going. Let's stick to this separate type, but I'm thinking we call it `BackstageTypography` to stick with the pattern of the other exports